doc/src/explanation/best-practices.md

```diff
@@ -0,0 +1,1329 @@
+  Python, `"(x + y)/z"` is no different from `"/x+)(y z "`, which is the same
+  string with the characters in another order. But SymPy expressions know what
+  type of expression they are and how to traverse the expression tree (e.g.,
+  using `expr.args`), and you build then up with operators like `+`, `-`, `*`,
+  or `/`. And to the previous point, if there is an error, it will be
+  difficult to track down because it won't be noticed until you actually parse
+  the string into an expression.
+
+  For example
+
+  **Don't**
+
+  ```py
+  >>> expression_str = '+'.join([f'{i}*x_{i}' for i in range(10)])
+  >>> expr = parse_expr(expression_str)
+  >>> expr
+  x_1 + 2*x_2 + 3*x_3 + 4*x_4 + 5*x_5 + 6*x_6 + 7*x_7 + 8*x_8 + 9*x_9
+  ```
+
+  **Do**
+
+  ```py
+  >>> from sympy import Add, Symbol
+  >>> expr = Add(*[i*Symbol(f'x_{i}') for i in range(10)])
+  >>> expr
+  x_1 + 2*x_2 + 3*x_3 + 4*x_4 + 5*x_5 + 6*x_6 + 7*x_7 + 8*x_8 + 9*x_9
+  ```
+
+- In code editors that do syntax highlighting, strings will be highlighted all
+  one color, whereas Python expressions will be highlighted according to their
+  actual content.
+
+- Support for string inputs in general SymPy functions (other than `sympify()`
+  or `parse_expr()`) [may go away in a future version of
+  SymPy](https://github.com/sympy/sympy/issues/11003).
+
+(best-practices-exact-rational-numbers-vs-floats)=
+### Exact Rational Numbers vs. Floats
+
+If a number is known to be exactly equal to some quantity, avoid defining it
+as a floating-point number.
+
+For example,
+
+**Don't**
+
+```py
+>>> expression = x**2 + 0.5*x + 1
+```
+
+**Do**
+
+```py
+>>> from sympy import Rational
+>>> expression = x**2 + Rational(1, 2)*x + 1
+>>> expression = x**2 + x/2 + 1 # Equivalently
+```
+
+However, this isn't to say that you should never use floating-point numbers in
+SymPy, only that if a more exact value is known it should be preferred. SymPy
+does support [arbitrary precision floating-point
+numbers](sympy.core.numbers.Float), but some operations may not perform as
+well with them.
+
+This also applies to non-rational numbers which can be represented exactly. For
+example, one should avoid using `math.pi` and prefer `sympy.pi`, since the
+former is a numerical approximation to $\pi$ and the latter is exactly $\pi$
+(see also [](best-practices-separate-symbolic-and-numeric-code) below; in
+general, one should avoid importing `math` when using SymPy).
+
+**Don't**
+
+```py
+>>> import math
+>>> import sympy
+>>> math.pi
+3.141592653589793
+>>> sympy.sin(math.pi)
+1.22464679914735e-16
+```
+
+**Do**
+
+```py
+>>> sympy.pi
+pi
+>>> sympy.sin(sympy.pi)
+0
+```
+
+Here `sympy.sin(math.pi)` is not exactly 0, because `math.pi` is not exactly $\pi$.
+
+
+One should also take care to avoid writing `integer/integer` where both
+integers are explicit integers. This is because Python will evaluate this to a floating-point value before SymPy is able to parse it.
```

Comment:
```suggestion
integers are explicit integers. This is because Python will evaluate
this to a floating-point value before SymPy is able to parse it.
```

Reply:
I'm having difficulty reading the GitHub diff. Is there an actual change to the text here or is this just a line wrap?